test/task_running_resources_test.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import time
+import signal
+import multiprocessing
+from contextlib import contextmanager
+
+from helpers import unittest, RunOnceTask
+
+import luigi
+import luigi.server
+
+
+class ResourceTestTask(RunOnceTask):
+
+    param = luigi.Parameter()
+    reduce_foo = luigi.BoolParameter()
+
+    def process_resources(self):
+        return {"foo": 2}
+
+    def run(self):
+        if self.reduce_foo:
+            self.decrease_running_resources({"foo": 1})
+
+        time.sleep(2)
+
+        super(ResourceTestTask, self).run()
+
+
+class ResourceWrapperTask(RunOnceTask):
+
+    reduce_foo = ResourceTestTask.reduce_foo
+
+    def requires(self):
+        return [
+            ResourceTestTask(param="a", reduce_foo=self.reduce_foo),
+            ResourceTestTask(param="b"),
+        ]
+
+
+class LocalRunningResourcesTest(unittest.TestCase):
+
+    def test_resource_reduction(self):
+        # trivial resource reduction on local scheduler
+        # test the running_task_resources setter and getter
+        sch = luigi.scheduler.Scheduler(resources={"foo": 2})
+
+        with luigi.worker.Worker(scheduler=sch) as w:
+            task = ResourceTestTask(param="a", reduce_foo=True)
+
+            w.add(task)
+            w.run()
+
+            self.assertEqual(sch.get_running_task_resources(task.task_id)["resources"]["foo"], 1)
+
+
+class ConcurrentRunningResourcesTest(unittest.TestCase):
+
+    def get_app(self):
```

Comment:
Can you remove this in a follow-up? You don't need it anymore right?

Reply:
Yup!